package.json

```diff
@@ -127,6 +127,7 @@
     "@emotion/styled": "^11.8.1",
     "@mui/material": "^5.6.2",
     "@types/deep-equal": "^1.0.1",
+    "@types/lodash": "^4.14.191",
```

Comment:
I see you've added `lodash` for the `isEqual` function.  We already have `deep-equal` as a dependency (see the line above this change).  Could that work so we avoid added another dependency?

Reply:
If `deep-equal` doesn't work you probably then want to see if you can only import just `isEqual` from lodash where you use it instead of importing the whole library.